src/cross_sections.cpp

```diff
@@ -0,0 +1,433 @@
+      int n_e = data::ttb_e_grid.size();
+      data::energy_min[photon] = std::max(data::energy_min[photon], data::ttb_e_grid(1));
+      data::energy_max[photon] = std::min(data::energy_max[photon], data::ttb_e_grid(n_e - 1));
+    }
+
+    // Take logarithm of energies since they are log-log interpolated
+    data::ttb_e_grid = xt::log(data::ttb_e_grid);
+  }
+
+  // Show which nuclide results in lowest energy for neutron transport
+  for (const auto& nuc : data::nuclides) {
+    // If a nuclide is present in a material that's not used in the model, its
+    // grid has not been allocated
+    if (nuc->grid_.size() > 0) {
+      double max_E = nuc->grid_[0].energy.back();
+      int neutron = static_cast<int>(Particle::Type::neutron);
+      if (max_E == data::energy_max[neutron]) {
+        write_message("Maximum neutron transport energy: " +
+          std::to_string(data::energy_max[neutron]) + " eV for " +
+          nuc->name_, 7);
+        if (mpi::master && data::energy_max[neutron] < 20.0e6) {
+          warning("Maximum neutron energy is below 20 MeV. This may bias "
+            " the results.");
+        }
+        break;
+      }
+    }
+  }
+
+  // Show minimum/maximum temperature
+  write_message("Minimum neutron data temperature: " +
+    std::to_string(data::temperature_min) + " K", 4);
+  write_message("Maximum neutron data temperature: " +
+    std::to_string(data::temperature_max) + " K", 4);
+
+  // If the user wants multipole, make sure we found a multipole library.
+  if (settings::temperature_multipole) {
+    bool mp_found = false;
+    for (const auto& nuc : data::nuclides) {
+      if (nuc->multipole_) {
+        mp_found = true;
+        break;
+      }
+    }
+    if (mpi::master && !mp_found) {
+      warning("Windowed multipole functionality is turned on, but no multipole "
+        "libraries were found. Make sure that windowed multipole data is "
+        "present in your cross_sections.xml file.");
+    }
+  }
+}
+
+void read_ce_cross_sections_xml()
+{
+  // Check if cross_sections.xml exists
+  const auto& filename = settings::path_cross_sections;
+  if (!file_exists(filename)) {
+    // Could not find cross_sections.xml file
+    fatal_error("Cross sections XML file '" + filename +
+      "' does not exist.");
+  }
+
+  write_message("Reading cross sections XML file...", 5);
+
+  // Parse cross_sections.xml file
+  pugi::xml_document doc;
+  auto result = doc.load_file(filename.c_str());
+  if (!result) {
+    fatal_error("Error processing cross_sections.xml file.");
+  }
+  auto root = doc.document_element();
+
+  std::string directory;
+  if (check_for_node(root, "directory")) {
+    // Copy directory information if present
+    directory = get_node_value(root, "directory");
+  } else {
+    // If no directory is listed in cross_sections.xml, by default select the
+    // directory in which the cross_sections.xml file resides
+    auto pos = filename.rfind("/");
+    if (pos == std::string::npos) {
```

Comment:
A real platform check would be better here.

Reply:
`std::filesystem` would make this straightforward, but that's C++17 which we don't use yet. I'll put a TODO to use it when C++17 is adopted